server/util-server.js

```diff
@@ -70,3 +70,53 @@ exports.getSettings = async function (type) {
 
     return result;
 }
+
+
+// ssl-checker by @dyaa
+// param: res - response object from axios
+// return an object containing the certificate information
+
+const getDaysBetween = (validFrom, validTo) =>
+    Math.round(Math.abs(+validFrom - +validTo) / 8.64e7);
+
+const getDaysRemaining = (validFrom, validTo) => {
+    const daysRemaining = getDaysBetween(validFrom, validTo);
+    if (new Date(validTo).getTime() < new Date().getTime()) {
+        return -daysRemaining;
+    }
+    return daysRemaining;
+};
+
+exports.checkCertificate = function (res) {
+    const {
+        valid_from,
+        valid_to,
+        subjectaltname,
+        issuer,
+        fingerprint,
+    } = res.request.res.socket.getPeerCertificate(false);
```

Comment:
I just test the code, the problem is strange.

After started server, all monitor with https are able to get the info on the first heartbeat.

However, for the second heartbeat, res.request.res.socket.getPeerCertificate(false) return empty object {}. So it goes into "No certificate" error.

I am testing on Windows 10 (x64) Node.js 14.16.0

Reply:
Sorry about that, apparently session reuse will cause no certificate to be returned. Fixed in new commit.